lms/static/js/spec/student_account/account_settings_fields_spec.js

```diff
@@ -0,0 +1,337 @@
+                view = new AccountSettingsFieldViews.DropdownFieldView(fieldData).render();
+                expectTitleAndMessageToBe(view, fieldData.title, fieldData.helpMessage);
+                expect(view.$('.account-settings-field-value option:selected').val()).toBe('');
+                expect(view.$('.account-settings-field-value option:selected').text().trim()).toBe('');
+            });
+
+            it("can correctly render link field", function() {
+                var fieldData = createFieldData(AccountSettingsFieldViews.LinkFieldView, {
+                    title: 'Title',
+                    linkTitle: 'Link title',
+                    helpMessage: 'Click the link.'
+                });
+                var view = new AccountSettingsFieldViews.LinkFieldView(fieldData).render();
+                expectTitleAndMessageToBe(view, fieldData.title, fieldData.helpMessage);
+                expect(view.$('.account-settings-field-value > a').text().trim()).toBe(fieldData.linkTitle);
+            });
+
+            it("can correctly render the text field help messages", function() {
+                var textFieldSelector = '.account-settings-field-value > input';
+                var messageSelector = '.account-settings-field-message';
+                var data = {'name': 'Legolas Thranduil'};
+                var fieldData = createFieldData(AccountSettingsFieldViews.TextFieldView, {
+                    title: 'Full Name',
+                    valueAttribute: 'name',
+                    helpMessage: 'edX full name'
+                });
+                var textFieldView = new AccountSettingsFieldViews.TextFieldView(fieldData).render();
+                requests = AjaxHelpers.requests(this);
+
+                // Verify default message
+                expect(textFieldView.$(messageSelector).text().trim()).toBe(fieldData.helpMessage);
+
+                // Verify change in-progress message
+                textFieldView.$(textFieldSelector).trigger('change');
+                expect(textFieldView.$(messageSelector).text().trim()).toBe('Saving...');
+
+                // Verify change completion message
+                textFieldView.$(textFieldSelector).trigger('change');
+                fieldValueChangeEventHandler(this, data);
+                expect(textFieldView.$(messageSelector).text().trim()).toBe('Successfully changed.');
+
+                // Verify server side issue/failure message
+                textFieldView.$(textFieldSelector).val(data.name).change();
+                fieldValueChangeEventHandler(this, data, 500);
+                expect(textFieldView.$(messageSelector).text().trim()).toBe('An error occurred, please try again.');
+
+                // Verify wrong input message
+                // Empty the text field which is not a valid value, so we should get an error message
+                textFieldView.errorMessage = '<i class="fa fa-exclamation-triangle"></i> Invalid Full Name value.';
```

Comment:
add an ```aria-hidden=true``` attribute on the icon font and a screen reader accessible label i.e. ```<span class="sr">Error:</span>``` as  a sibling to the icon font (between the icon and the actual error msg)